backend/app/mcp/tools.py

```
@@ -0,0 +1,466 @@
+        if not 50 <= sample_size <= 10000:
+            raise ToolError("sample_size must be between 50 and 10000")
+        if not 5 <= batch_size <= 100:
+            raise ToolError("batch_size must be between 5 and 100")
+
+        async def progress_cb(p: int, msg: str | None = None) -> None:
+            await ctx.report_progress(progress=p, total=100)
+            if msg:
+                try:
+                    await ctx.info(msg)
+                except Exception:  # noqa: BLE001 — info 실패가 분석을 막지 않음
+                    pass
```

Ruff (0.15.12)

[warning] Use `contextlib.suppress(Exception)` instead of `try`-`except`-`pass`

(SIM105)